generated/nifake/unit_tests/test_session.py

```diff
@@ -158,6 +159,28 @@ def test_simple_function(self):
             session.simple_function()
             self.patched_library.niFake_PoorlyNamedSimpleFunction.assert_called_once_with(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST))
 
+    def test_self_test(self):
+        self.patched_library.niFake_self_test.side_effect = self.side_effects_helper.niFake_self_test
+        test_error_code = 0
+        self.side_effects_helper['self_test']['selfTestResult'] = test_error_code
+        self.side_effects_helper['self_test']['selfTestMessage'] = ''
+        with nifake.Session('dev1') as session:
+            session.self_test()
+
+    def test_self_test_fail(self):
+        self.patched_library.niFake_self_test.side_effect = self.side_effects_helper.niFake_self_test
+        test_error_code = 1
+        test_error_message = 'error message'
+        self.side_effects_helper['self_test']['selfTestResult'] = test_error_code
+        self.side_effects_helper['self_test']['selfTestMessage'] = test_error_message
+        with nifake.Session('dev1') as session:
+            try:
+                session.self_test()
+                assert False
+            except nifake.errors.SelfTestFailureError as e:
+                assert e.code == test_error_code
+                assert e.message == test_error_message
```

Comment:
You need to fail the test when an exception other than nifake.errors.SelfTestFailureError is thrown as well. As written we would get a false positive.